### PR TITLE
Adds visually hidden text to inline external links

### LINKED ISSUE #428: External links - appending text for screen readers
We need to apply text to external inline links that are read out by a screen reader. This text will be `, external`, i.e. `[TEXT], external`.

**Describe the solution you'd like**
Update the Inline Link component `@bbc/psammead-inline-link` to have a prop `isExternal` and when the value is `true`. This should add the correct Aria label: 

An aria-label should be used to indicate an external link to screen readers and the best support to achieve a pause is to use a comma in the code
```
<a href="https://www.bbc.co.uk/my-url" aria-label="Israel, external">Israel</a>
```

We should update the readme to add an explanation of this change in the Accessibility section. 

**Describe alternatives you've considered**
We considered adding Cypress integration tests for this. So a prerequisite for doing this work would be to update fixture data locally to include an article with an external link. However, we think that unit tests should be sufficient. 

**Testing notes**
Regression testing of the article using VoiceOver, TalkBack and on Desktop (JAWS).

Dev insight: 
- Unit tests should be added. 
- Use a screen reader to confirm that the label is read out as expected
https://bbc-news.github.io/accessibility-news-and-you/accessibility-and-testing-with-assistive-technology

**Additional context**

- [x] Initially labelled with ["Refinement needed"](https://github.com/bbc/simorgh/labels/Refinement%20Needed)

### PR DESCRIPTION
**Pull request by ChrisBAshton**
Resolves #428

**Overall change:** Checks the `isExternal` boolean in the `urlBlock` data. If `true`, appends a visually hidden `, external` string to the inline link. The effect being that this text is hidden but read out to screen reader users, e.g. a link that visually renders as "Visit Google" would be announced as "Visit Google, external".

External link example: http://localhost:7080/news/articles/c0000000006o
Internal link example: http://localhost:7080/news/articles/c85pqyj5m2ko

**Code changes:**

- Updates `InlineLink` container to have a required `isExternal` boolean.
- Adds unit test checking the shallow snapshot render if a link is external.
- Updates affected snapshots.
- Offscreen text is localised - so works for Persian and English, using the Service Context. Note, in future, when we could have multiple languages per service, we need to split this out into a Language Context and a Service Context.
- Adding Storybook stories for internal, external links for the News and Persian service. 

Offscreen text here:
screen shot offscreen text

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval
- [ ] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

### CODE REVIEW

File: src/app/containers/InlineLink/index.jsx [hunk 1]
```diff
@@ -29,6 +30,10 @@ const InlineLinkContainer = ({ locator, blocks }) => {
     );
   }
 
+  if (isExternal) {
+    blocks.push(<VisuallyHiddenText>, external</VisuallyHiddenText>);
```
Comment (dr3) [inline src/app/containers/InlineLink/index.jsx]: I dont think this is right, the blocks array contains objects of type and model https://github.com/bbc/simorgh/blob/latest/src/app/containers/Blocks/index.jsx#L30-L35

**Review by dr3 — CHANGES_REQUESTED** [review]

File: src/app/containers/InlineLink/index.jsx (hunk 1 above)
Comment (ChrisBAshton) [reply src/app/containers/InlineLink/index.jsx]: Argh, good point!

File: .env [hunk 2]
```diff
@@ -1,4 +1,6 @@
-# On running the 'build' commands this file is replaced by the files in `env-config/`
-# To ignore and keep this file we have a command in `postshrinkwrap` which assumes this file is not changing
-# The command is `git update-index --assume-unchanged .env` 
-# Source: https://wildlyinaccurate.com/git-ignore-changes-in-already-tracked-files/
+SIMORGH_BASE_URL=http://localhost:7080
```
Comment (dr3) [inline .env]: This file needs to be reset to latest 

@pjlee11 Looks like your git trick isn't working, can you maybe raise an issue to take a look at it please?

File: src/app/containers/InlineLink/index.jsx [hunk 3]
```diff
@@ -33,6 +34,7 @@ const InlineLinkContainer = ({ locator, blocks }) => {
   return (
     <InlineLink href={locator}>
       <Blocks blocks={blocks} componentsToRender={componentsToRender} />
+      {isExternal ? <VisuallyHiddenText>, external</VisuallyHiddenText> : null}
```
Comment (dr3) [inline src/app/containers/InlineLink/index.jsx]: 👍

File: src/app/containers/Figure/fixtureData.jsx [hunk 4]
```diff
@@ -38,6 +38,7 @@ const captionBlockWithLink = {
                     model: {
                       text: 'containing an inline link',
                       locator: 'https://www.bbc.com',
+                      isExternal: false,
```
Comment (dr3) [inline src/app/containers/Figure/fixtureData.jsx:41]: Any chance we can get a 'true' case in fixture?

File: src/app/containers/Figure/fixtureData.jsx [hunk 5]
```diff
@@ -38,6 +38,7 @@ const captionBlockWithLink = {
...
                       blocks: [
                         {
```
Comment (dr3) [inline src/app/containers/Figure/fixtureData.jsx:43]: Can we maybe add Inline link to the Simorgh Storybook now that its some unique container logic please? One external and internal would be nice

**Review by dr3 — CHANGES_REQUESTED** [review]
Looking gooood

File: src/app/containers/Figure/fixtureData.jsx (hunk 4 above)
Comment (sareh) [reply src/app/containers/Figure/fixtureData.jsx:41]: Added relevant attribute into http://localhost:7080/news/articles/c0000000006o, which had an external link.

File: src/app/containers/Figure/fixtureData.jsx (hunk 5 above)
Comment (sareh) [reply src/app/containers/Figure/fixtureData.jsx:43]: These have been added.

File: src/app/containers/InlineLink/__snapshots__/index.test.jsx.snap [hunk 6]
```diff
@@ -1,6 +1,152 @@
...
+        },
+      ]
+    }
+    isExternal={true}
+    locator="https://www.example.com/"
+  />
+</Context.Provider>
+`;
+
+exports[`InlineLinkContainer external link accessibility should be explicitly marked "external" for screen reader users 1`] = `
```
Comment (dr3) [inline src/app/containers/InlineLink/__snapshots__/index.test.jsx.snap:53]: I dont think this snapshot is testing what we want, might need to do a deep render

File: src/app/containers/InlineLink/__snapshots__/index.test.jsx.snap [hunk 7]
```diff
@@ -1,6 +1,152 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
-exports[`InlineLinkContainer with link matching routes for SPA should render correctly 1`] = `
+exports[`InlineLinkContainer external link accessibility should be explicitly marked "external" for screen reader users & localised 1`] = `
```
Comment (dr3) [inline src/app/containers/InlineLink/__snapshots__/index.test.jsx.snap:3]: or this one :(

File: src/app/containers/InlineLink/index.stories.jsx [hunk 8]
```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react'; // eslint-disable-line import/no-extraneous-dependencies
+import InlineLinkContainer from '.';
+import { ServiceContextProvider } from '../../contexts/ServiceContext';
+
+// const InternalInlineLink = (locator, blocks, isExternal) => {
```
Comment (dr3) [inline src/app/containers/InlineLink/index.stories.jsx]: ??

**Review by dr3 — CHANGES_REQUESTED** [review]

File: src/app/lib/config/services/persian.js [hunk 9]
```diff
@@ -5,6 +5,7 @@ const persian = {
   brandName: 'BBC News فارسی',
   defaultImage: 'https://news.files.bbci.co.uk/ws/img/logos/og/persian.png',
   defaultImageAltText: 'BBC News فارسی',
+  externalLinkText: ' ،لینک خارجی',
```
Comment (dr3) [inline src/app/lib/config/services/persian.js:8]: 🎉

**Review by dr3 — APPROVED** [review]
👍 LGTM, Great work :D

**Review by twinlensreflex — APPROVED** [review]
tested locally and it lgtm! 👍

**Comment by sareh** [comment]
VoiceOver reads out the external link example as `dui neque consectetur magna, external`.

File: src/app/containers/InlineLink/index.jsx [hunk 10]
```diff
@@ -31,9 +33,16 @@ const InlineLinkContainer = ({ locator, blocks }) => {
...
-    <InlineLink href={locator}>
-      <Blocks blocks={blocks} componentsToRender={componentsToRender} />
-    </InlineLink>
+    <ServiceContextConsumer>
+      {({ externalLinkText }) => (
+        <InlineLink href={locator}>
+          <Blocks blocks={blocks} componentsToRender={componentsToRender} />
+          {isExternal ? (
+            <VisuallyHiddenText>{externalLinkText}</VisuallyHiddenText>
+          ) : null}
```
Comment (ChrisBAshton) [inline src/app/containers/InlineLink/index.jsx:42]: Based on Psammead meeting today - `externalLinkText` will be a prefix on Persian and a suffix on News, right? 🤔

So we've i18ned the text here but not the placement of it 😅

Comment (sareh) [reply src/app/containers/InlineLink/index.jsx:42]: Yes, agreed. We have several examples of containers with translated text. In Brand, Caption, Copyright, ErrorMain. 
`brandName`
`imageCaptionOffscreenText`
`imageCopyrightOffscreenText`

**Comment by greenc05** [comment]
@ChrisBAshton aria-label was the preferred implementation choice mainly because it won't add the text ', external' to the article text. Using the visually hidden approach, this text will be seen when CSS doesn't load/or is turned off. https://docs.google.com/document/d/1AmOcf7e4njuUlBlV8ZZXVTto3Cd68mCfzkZX6jBI_-U/edit?usp=sharing

**Pull request merged**